abc.templates/deployments/contents/workflows/deploy-github-token-minter.yaml

```diff
@@ -0,0 +1,53 @@
+name: 'deploy to github-token-minter'
+
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - 'REPLACE_SUBDIRECTORY/**'
+  workflow_dispatch:
+
+env:
+  WIF_PROVIDER: 'REPLACE_WIF_PROVIDER'
+  WIF_SERVICE_ACCOUNT: 'REPLACE_WIF_SERVICE_ACCOUNT'
+
+
+# Don't cancel in progress since we don't want to have half-baked releases.
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}-release'
+
+permissions:
+  contents: 'read'
+  id-token: 'write'
+
+jobs:
+  release_and_deploy:
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: 'Checkout'
+        uses: 'actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744' # ratchet:actions/checkout@v3
+
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@35b0e87d162680511bf346c299f71c9c5c379033' # ratchet:google-github-actions/auth@v1
+        with:
+          workload_identity_provider: '${{ env.WIF_PROVIDER }}'
+          service_account: '${{ env.WIF_SERVICE_ACCOUNT }}'
+          token_format: 'access_token'
+
+      - name: 'Setup gcloud'
+        uses: 'google-github-actions/setup-gcloud@e30db14379863a8c79331b04a9969f4c1e225e0b' # ratchet:google-github-actions/setup-gcloud@v1
+
+      - uses: 'docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc' # ratchet:docker/login-action@v2
```

Comment:
Something to consider here: the authentication token is usually only valid for a few minutes (5min by default). If the build step takes more than 5 minutes, then the auth token provided to `docker login` will have expired and the push will fail. It'd be more full-proof to do something like:

- auth
- docker login (so `FROM` tags can use private registries)
- docker build
- auth (again, get a new token)
- docker login (again with the new token)
- docker push

Reply:
That is a valid point but I expect a short build time for bootstrapping because we're only adding configuration files on top of the parent image. The intention of having everything in `deploy.sh` is to more easily identify the version of github-token-minter, but if the build step is a concern then we should separate it accordingly to your full-proof solution.

Reply:
The idea here was for readability, the source of truth is the deploy script. Lets keep this as-is and monitor build times. I dont see this surpassing 5 min, at least in the near term.